plugins/woocommerce/src/Utilities/NumberUtil.php

```diff
@@ -49,4 +53,66 @@ public static function array_sum( array $arr ): float {
 
 		return array_sum( $sanitized_array );
 	}
+
+	/**
+	 * Sanitize a cost value based on the current locale decimal and thousand separators.
+	 *
+	 * @param string $value               The value to sanitize.
+	 * @return string The sanitized value, or empty string if invalid.
+	 * @throws \InvalidArgumentException If the value is not a valid numeric value.
+	 */
+	public static function sanitize_cost_in_current_locale( $value ): string {
+		$value                      = is_null( $value ) ? '' : $value;
+		$value                      = wp_kses_post( trim( wp_unslash( $value ) ) );
+		$currency_symbol_encoded    = get_woocommerce_currency_symbol();
+		$currency_symbol_variations = array( $currency_symbol_encoded, wp_kses_normalize_entities( $currency_symbol_encoded ), html_entity_decode( $currency_symbol_encoded ) );
+
+		$value = str_replace( $currency_symbol_variations, '', $value );
+
+		$allowed_characters_regex = sprintf(
+			'/^[0-9\%s\%s]*$/',
+			wc_get_price_thousand_separator(),
+			wc_get_price_decimal_separator()
+		);
+
+		if ( 1 !== preg_match( $allowed_characters_regex, $value ) ) {
+			/* translators: %s: Invalid value that was input by the user */
+			throw new \InvalidArgumentException( sprintf( esc_html__( '%s is not a valid numeric value. Allowed characters are numbers and the thousand and decimal separators.', 'woocommerce' ), esc_html( $value ) ) );
+		}
+
+		// Validate decimal and thousand separator positions.
+		$decimal_separator  = wc_get_price_decimal_separator();
+		$thousand_separator = wc_get_price_thousand_separator();
+
+		if (
+			// Check that there is only 1 decimal separator.
+			substr_count( $value, $decimal_separator ) > 1 ||
+			(
+				// Check that decimal separator appears after thousand separator if both exist.
+				str_contains( $value, $thousand_separator ) &&
+				str_contains( $value, $decimal_separator ) &&
```

Comment:
`str_contains` is only available in PHP 8+.

```suggestion
				false !== strpos( $value, $thousand_separator ) &&
				false !== strpos( $value, $decimal_separator ) &&
```